smtp_test.go

```diff
@@ -239,6 +257,55 @@ func testSendMailTimeout(t *testing.T, d *Dialer, want []string) {
 	doTestSendMail(t, d, want, true)
 }
 
+func doTestSMTPReset(t *testing.T, d *Dialer, want []string){
+	testClient := &mockClient{
+		t:       t,
+		want:    want,
+		addr:    addr(d.Host, d.Port),
+		config:  d.TLSConfig,
+		timeout: false,
+	}
+
+	netDialTimeout = func(network, address string, d time.Duration) (net.Conn, error) {
```

Comment:
Looking at the build error, the problem is here. If you capitalize `NetDialTimeout` that should fix it. (We exported this in #4).